src/module/actor/initiative.ts

```diff
@@ -31,25 +31,10 @@ class ActorInitiative {
         this.statistic = statistic;
     }
 
-    async roll(args: InitiativeRollParams): Promise<InitiativeRollResult | null> {
+    async roll(args: InitiativeRollParams = {}): Promise<InitiativeRollResult | null> {
+        const updateTracker = args.updateTracker ?? true;
         // Get or create the combatant
-        const combatant = await (async (): Promise<CombatantPF2e<EncounterPF2e> | null> => {
-            if (!game.combat) {
-                ui.notifications.error(game.i18n.localize("PF2E.Encounter.NoActiveEncounter"));
-                return null;
-            }
-            const token = this.actor.getActiveTokens().pop();
-            const existing = game.combat.combatants.find((combatant) => combatant.actor === this.actor);
-            if (existing) {
-                return existing;
-            } else if (token) {
-                await token.toggleCombat(game.combat);
-                return token.combatant ?? null;
-            } else {
-                ui.notifications.error(game.i18n.format("PF2E.Encounter.NoTokenInScene", { actor: this.actor.name }));
-                return null;
-            }
-        })();
+        const combatant = await CombatantPF2e.fromActor(this.actor, updateTracker);
```

Comment:
The point of having the render option is so that we don't render twice, which will happen here.